system/HTTP/Cookie/CookieInterface.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP\Cookie;
+
+/**
+ * Interface for a value object representation of an HTTP cookie.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+ */
+interface CookieInterface
+{
+	/**
+	 * Cookies will be sent in all contexts, i.e in responses to both
+	 * first-party and cross-origin requests. If `SameSite=None` is set,
+	 * the cookie `Secure` attribute must also be set (or the cookie will be blocked).
+	 */
+	public const SAMESITE_NONE = 'none';
```

Comment:
the const are intentionally titled lowercase to aid validation. But if you'll check the implementation of how it is set in setcookie it is `ucfirst(strtolower($samesite))`

Reply:
@paulbalandan cant find `ucfirst(strtolower($samesite))` + this look like strange/unlogic.
you dont use self added constants:
`protected $samesite = 'Lax';
i cant find other reasons for hard code like this:
`if (! in_array(strtolower($config->cookieSameSite), array_merge(Cookie::ALLOWED_SAMESITE_VALUES, ['']), true))`

Reply:
> @paulbalandan cant find ucfirst(strtolower($samesite))

In `Cookie` class, look at lines 289, 450 and 770 for the `ucfirst(strtolower())`

> you dont use self added constants:
> `protected $samesite = 'Lax';`

Probably missed that part for updating. Thanks I'll change that.

> i cant find other reasons for hard code like this:
> `if (! in_array(strtolower($config->cookieSameSite), array_merge(Cookie::ALLOWED_SAMESITE_VALUES, ['']), true))`

When I initially modified this line, I was aiming support for PHP 7.2, but since we've already upgraded PHP requirement to 7.3, I'll try to have a more elegant solution to this.